geoips/commandline/geoips_run.py

```diff
@@ -3,36 +3,89 @@
 
-from geoips.commandline.geoips_command import GeoipsExecutableCommand
 
+class GeoipsRunConfigBased(GeoipsExecutableCommand):
+    """Run Sub-Command for executing the config based process-workflow (procflow)."""
 
-class GeoipsRun(GeoipsExecutableCommand):
-    """Run Sub-Command for running process-workflows (procflows)."""
+    command_name = "config_based"
+    subcommand_classes = []
 
-    command_name = "run"
+    def add_arguments(self):
+        """Add arguments to the run-subparser for the 'run config_based' Command."""
+        add_args(parser=self.subcommand_parser, legacy=self.legacy)
+
+    def __call__(self, args):
+        """Run the provided GeoIPS command.
+
+        Parameters
+        ----------
+        args: Namespace()
+            - The argument namespace to parse through
+        """
+        if args.procflow is None and self.legacy:
+            err_str = (
+                "Deprecated, Legacy 'run_procflow' call was used and --procflow "
+                "flag wasn't specified. Please either specify which procflow is "
+                "being executed via '--procflow <procflow_name>' or use the "
+                "supported procflow call 'geoips run <procflow_name>'"
+            )
+            self.subcommand_parser.error(err_str)
+        elif args.procflow is None:
+            # If None, set to 'config_based'. We don't want users to have to specify
+            # what procflow will be used as it is specified in
+            # 'geoips run config_based'.
+            args.procflow = "config_based"
+        main(ARGS=args)
+
+
+class GeoipsRunDataFusion(GeoipsExecutableCommand):
+    """Run Sub-Command for executing the data fusion process-workflow (procflow)."""
+
+    command_name = "data_fusion"
+    subcommand_classes = []
+
+    def add_arguments(self):
+        """Add arguments to the run-subparser for the 'run data_fusion' Command."""
+        data_fusion_add_args(parser=self.subcommand_parser, legacy=self.legacy)
+
+    def __call__(self, args):
+        """Run the provided GeoIPS command.
+
+        Parameters
+        ----------
+        args: Namespace()
+            - The argument namespace to parse through
+        """
+        if args.procflow is None and self.legacy:
+            err_str = (
+                "Deprecated, Legacy 'data_fusion_procflow' call was used and "
+                "--procflow flag wasn't specified. Please either specify which "
+                "procflow is being executed via '--procflow <procflow_name>' or "
+                "use the supported procflow call 'geoips run <procflow_name>'"
+            )
+            self.subcommand_parser.error(err_str)
+        elif args.procflow is None:
+            # If None, set to 'data_fusion'. We don't want users to have to specify
+            # what procflow will be used as it is specified in
+            # 'geoips run data_fusion'.
+            args.procflow = "data_fusion"
+        main(ARGS=args)
```

Comment:
Huh, that's confusing. I didn't realize that `data_fusion_procflow` actually called `main` from `run_procflow`. That seems pretty circuitous... Not worth fixing, just saying it seems weird.

Reply:
Agreed that it's weird. I just followed the way it was implemented currently, but I agree that it seems like a loop. Hopefully order_based will fix this.